src/services/postService.js

```diff
@@ -53,9 +79,33 @@ async function checkLike(like, postID, userID){
     return postData;
 }
 
+const deletePost = async (postId) => {
+    await getPostById(postId);
+
+    const deleteResult = await postDAO.deletePost(postId);
+    throwIfError(deleteResult);
+}
+
+const deleteReply = async (postId, replyId) => {
+    const repliesOfPost = await getRepliesOfPost(postId);
+    
+    const index = repliesOfPost.findIndex((reply) => reply.itemID === replyId);
+    if (index === -1) {
+        throw { status: 400, message: "That reply doesn't exist" }
+    }
+    
+    const newReplies = repliesOfPost.filter((reply) => reply.itemID !== replyId); // tried using splice() but for some reason does not work
+    const data = await postDAO.updateReplies(postId, newReplies);
+    throwIfError(data);
+}
+
 module.exports = {
     createPost,
     createReply,
+    getPostById,
     seePosts,
-    checkLike
+    getReplyOfPost,
```

Comment:
I would remove getReplyOfPost from exports. It looked like a helper function to get a reply/index of a reply for you to delete a reply so it shouldn't be needed anywhere else

Reply:
I used it in authMiddleware.

Reply:
I also needed to use it in authMiddleware.